homeassistant/components/snmp/switch.py

```diff
@@ -277,6 +277,12 @@ async def async_update(self) -> None:
                 ):
                     self._state = False
                 else:
+                    _LOGGER.warning(
+                        "SNMP warning: "
+                        "did not recognise %s as a valid response payload, "
+                        "state will be unknown",
+                        resrow[-1],
+                    )
```

Comment:
The logger will write the module name and warning as part of the logging already, so we do not need to repeat that. Let's add the entity_id so the user knows which switch has the problem. Perhaps the message looks like this

```
"Invalid payload '%s' received for entity %s state is unknown"
```

Reply:
`WARNING (MainThread) [homeassistant.components.snmp.switch] Invalid payload '1' received for entity switch.9858dx_outlet_4, state is unknown`